libs/ui-lib/lib/ocm/hooks/useUISettings.ts

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { Cluster } from '@openshift-assisted/types/./assisted-installer-service';
+import { UISettingService, UISettingsValues } from '../services';
+
+const useUISettings = (clusterId?: Cluster['id']) => {
+  const [uiSettings, setUISettings] = React.useState<UISettingsValues>();
+
+  const updateUISettings = React.useCallback(
+    async (newSettings: UISettingsValues) => {
+      if (clusterId) {
```

Comment:
Making `clusterId` mandatory would make this check unnecessary

Reply:
The problem is that I'm using this hook in the `ClusterWizardContext` where cluster can be `undefined`. (e.g. on the New cluster page - we don't have a cluster yet, but we do have a wizard).

So I can make the `clusterId` mandatory in the service and API, but I cannot make it mandatory in this hook.